src/traits.rs

```diff
@@ -0,0 +1,19 @@
+use std::error::Error;
+use std::fmt::Display;
+use std::process::ChildStdout;
+
+pub trait Runnable {
```

Comment:
Are traits just templates?

Reply:
I think they're more equivalent to Interfaces (e.g. in Java). The book has a section on it [here](https://doc.rust-lang.org/book/ch10-02-traits.html).